src/commands/path/path.generate.ts

```diff
@@ -0,0 +1,192 @@
+    return `s3://${metadata.targetBucketName}/${metadata.region}/${name}_${metadata.date}_${metadata.gsd}m/rgb/${metadata.epsg}/`;
+  } else if ([dataCategories.DEM, dataCategories.DSM].includes(metadata.category)) {
+    return `s3://${metadata.targetBucketName}/${metadata.region}/${name}_${metadata.date}/${metadata.category}_${metadata.gsd}m/${metadata.epsg}/`;
+  } else {
+    throw new Error(`Path Can't be generated from collection as no matching category: ${metadata.category}.`);
+  }
+}
+
+function formatBucketName(bucketName: string): string {
+  if (bucketName.startsWith('s3://')) {
+    return bucketName.replace('s3://', '').replace('/', '');
+  }
+  return bucketName;
+}
+
+/**
+ * Generates specific dataset name based on metadata inputs
+ *
+ * @export
+ * @param {string} region
+ * @param {?string} [geographicDescription]
+ * @param {?string} [event]
+ * @returns {string}
+ */
+export function formatName(region: string, geographicDescription?: string, event?: string): string {
+  if (geographicDescription) {
+    return slugify([geographicDescription, event].filter(Boolean).join('-'));
+  }
+  return slugify([region, event].filter(Boolean).join('-'));
+}
+
+export function formatDate(collection: StacCollection): string {
+  const interval = collection.extent?.temporal?.interval?.[0];
+  const startYear = interval[0]?.slice(0, 4);
+  const endYear = interval[1]?.slice(0, 4);
+
+  if (startYear == null || endYear == null) {
+    throw new Error(`Missing datetime in interval: ${interval}`);
+  }
+  if (startYear === endYear) {
+    return startYear;
+  }
+  return `${startYear}-${endYear}`;
+}
+
+/*
+ *  nb: The following functions: 'loadFirstTiff', 'extractGsd', and 'extractEpsg' are
+ *  workarounds for use until the eo stac extension is added to the collection.json.
+ */
+
+/**
+ * Gets tiff of first item listed in collection
+ *
+ * @async
+ * @param {string} source
+ * @param {StacCollection} collection
+ * @returns {Promise<CogTiff>}
+ */
+export async function loadFirstTiff(source: string, collection: StacCollection): Promise<CogTiff> {
+  const itemLink = collection.links.find((f) => f.rel === 'item')?.href;
+  if (itemLink == null) throw new Error(`No items in collection from ${source}.`);
+  const itemPath = new URL(itemLink, source).href;
+  const item = await fsa.readJson<StacItem>(itemPath);
+  if (item == null) throw new Error(`Failed to get item.json from ${itemPath}.`);
+  const tiffLink = item.assets['visual']?.href;
+  if (tiffLink == null) throw new Error(`No tiff assets in Item: ${itemPath}`);
+  const tiffPath = new URL(tiffLink, source).href;
+  const tiff = await createTiff(tiffPath);
+  if (tiff == null) throw new Error(`Failed to get tiff from ${tiffPath}.`);
+  return tiff;
+}
+
+export function extractGsd(tiff: CogTiff): number {
+  const gsd = tiff.images[0]?.resolution[0];
+  if (gsd == null) {
+    throw new Error(`Missing resolution tiff tag`);
```

Comment:
NIT: generally its good practice to include the thing that caused the error, in this case the location to the tiff that doesnt have the resolution `tiff.source.url`

Reply:
I will add this in is own PR. for now it would be good to merge this